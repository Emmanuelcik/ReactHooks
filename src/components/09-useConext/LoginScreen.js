import React, { useContext } from 'react'
import { UserContext } from './UserContext';


export const LoginScreen = () => {
    
    const {user, setUser} = useContext(UserContext);
    console.log(user);
    const obj = {
        id: 1,
        nombre: "nose",
    }
    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>
            <button 
                className='btn btn-primary' 
                onClick={ () => {
                    setUser(obj);
                }}
            >
                Login
            </button>
        </div>
    )
}
