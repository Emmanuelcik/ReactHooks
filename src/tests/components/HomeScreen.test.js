import { mount } from "enzyme";
import { HomeScreen } from "../../components/09-useConext/HomeScreen";
import { UserContext } from "../../components/09-useConext/UserContext";


describe("Testing HomeScreen", () => {

    const user = {
        name: "Emm",
        email: "no se",
    }

    const wrapper = mount(
        <UserContext.Provider value={ {
            user,
        }}>

            <HomeScreen /> 
        </UserContext.Provider> 
    );
        
    test("Should match the snapshot", () => {

        expect(wrapper).toMatchSnapshot();
    })
});