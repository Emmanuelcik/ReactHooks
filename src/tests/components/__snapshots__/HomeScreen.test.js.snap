// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeScreen Should match the snapshot 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
   "name": "Emm",
   "email": "no se"
}
  </pre>
</div>
`;
